test/gtest/ucp/test_ucp_fence.cc

```diff
@@ -176,16 +203,41 @@ class test_ucp_fence : public test_ucp_atomic {
         status = ucp_rkey_pack(receiver().ucph(), memh, &rkey_buffer, &rkey_buffer_size);
         ASSERT_UCS_OK(status);
 
-        status = ucp_ep_rkey_unpack(sender().ep(), rkey_buffer, &rkey);
-        ASSERT_UCS_OK(status);
+        ucp_rkey_h *rkey = (ucp_rkey_h *)malloc(sizeof(ucp_rkey_h) * sender().get_num_workers());
```

Comment:
can use std::vector<ucp_rkey_h> here - would not need to worry about releasing it